clustered/server/src/main/java/org/ehcache/clustered/server/management/ClusterTierManagement.java

```diff
@@ -0,0 +1,132 @@
+  private final EhcacheStateService ehcacheStateService;
+  private final String storeIdentifier;
+
+  public ClusterTierManagement(ServiceRegistry services, EhcacheStateService ehcacheStateService, boolean active, String storeIdentifier) {
+    this.ehcacheStateService = ehcacheStateService;
+    this.storeIdentifier = storeIdentifier;
+
+    // create an entity monitoring service that allows this entity to push some management information into voltron monitoring service
+    EntityMonitoringService entityMonitoringService;
+    if (active) {
+      entityMonitoringService = services.getService(new ActiveEntityMonitoringServiceConfiguration());
+    } else {
+      IMonitoringProducer monitoringProducer = services.getService(new BasicServiceConfiguration<>(IMonitoringProducer.class));
+      entityMonitoringService = monitoringProducer == null ? null : services.getService(new PassiveEntityMonitoringServiceConfiguration(monitoringProducer));
+    }
+
+    // create a management registry for this entity to handle exposed objects and stats
+    // if management-server distribution is on the classpath
+    managementRegistry = entityMonitoringService == null ? null : services.getService(new ConsumerManagementRegistryConfiguration(entityMonitoringService));
+
+    if (managementRegistry != null) {
+
+      if (active) {
+        // expose settings about attached stores
+        managementRegistry.addManagementProvider(new ClusteredTierStateSettingsManagementProvider());
+      }
+
+      // expose settings about server stores
+      managementRegistry.addManagementProvider(new ServerStoreSettingsManagementProvider());
+      // expose settings about pools
+      managementRegistry.addManagementProvider(new PoolSettingsManagementProvider());
+
+      // expose stats about server stores
+      managementRegistry.addManagementProvider(new ServerStoreStatisticsManagementProvider());
+      // expose stats about pools
+      managementRegistry.addManagementProvider(new PoolStatisticsManagementProvider(ehcacheStateService));
+    }
+  }
+
+  // the goal of the following code is to send the management metadata from the entity into the monitoring tree AFTER the entity creation
+  public void init() {
+    if (managementRegistry != null) {
+      LOGGER.trace("init({})", storeIdentifier);
+      ServerSideServerStore serverStore = ehcacheStateService.getStore(storeIdentifier);
+      ServerStoreBinding serverStoreBinding = new ServerStoreBinding(storeIdentifier, serverStore);
+      managementRegistry.register(serverStoreBinding);
+      ServerSideConfiguration.Pool pool = ehcacheStateService.getDedicatedResourcePool(storeIdentifier);
+      if (pool != null) {
+        managementRegistry.register(new PoolBinding(storeIdentifier, pool, PoolBinding.AllocationType.DEDICATED));
+      }
+      managementRegistry.refresh();
+      managementRegistry.pushServerEntityNotification(serverStoreBinding, EHCACHE_SERVER_STORE_CREATED.name());
+    }
```

Comment:
Perhaps it's worse checking at other places too.... It's possible there's other places. The idea is that
- register must happen first before a refresh
- and pushNotification must happen after a register.
- refresh and push could happen at the same time